test/Microsoft.AspNetCore.Razor.Tools.Test/ServerCommandTest.cs

```diff
@@ -13,23 +13,23 @@ namespace Microsoft.AspNetCore.Razor.Tools
 {
     public class ServerCommandTest
     {
-        [Fact(Skip = "https://github.com/aspnet/Razor/issues/2310")]
+        [Fact]
         public void WritePidFile_WorksAsExpected()
         {
             // Arrange
             var expectedProcessId = Process.GetCurrentProcess().Id;
             var expectedRzcPath = typeof(ServerCommand).Assembly.Location;
             var expectedFileName = $"rzc-{expectedProcessId}";
-            var homeEnvVariable = PlatformInformation.IsWindows ? "USERPROFILE" : "HOME";
-            var path = Path.Combine(Environment.GetEnvironmentVariable(homeEnvVariable), ".dotnet", "pids", "build", expectedFileName);
+            var directoryPath = Path.Combine(Path.GetTempPath(), "RazorTest", Guid.NewGuid().ToString());
+            var path = Path.Combine(directoryPath, expectedFileName);
 
             var pipeName = Guid.NewGuid().ToString();
             var server = GetServerCommand(pipeName);
 
             // Act & Assert
             try
             {
-                using (var _ = server.WritePidFile())
+                using (var _ = server.WritePidFile(directoryPath))
```

Comment:
@pranavkm, I didn't need the Func<string, string> because we are passing the path directly.

Reply:
That was to verify the path calculation. Would be good add a test to cover that too

Reply:
Added.